sqlglot/dialects/bigquery.py

```diff
@@ -521,7 +537,8 @@ class Generator(generator.Generator):
             exp.TimestampSub: date_add_interval_sql("TIMESTAMP", "SUB"),
             exp.TimeStrToTime: timestrtotime_sql,
             exp.Trim: lambda self, e: self.func(f"TRIM", e.this, e.expression),
-            exp.TsOrDsAdd: date_add_interval_sql("DATE", "ADD"),
```

Comment:
was this incorrect?

Reply:
Yes, `date_add_interval_sql` doesn't add a cast and BQ doesn't like ts/ds in `DATE_ADD`

Reply:
So basically Hive `DATE_ADD` and `DATE_SUB` -> BigQuery would have this problem, as `TsOrDsAdd` is only used in `Hive` when parsing these functions.

Reply:
This is an example where the transpilation produced invalid BigQuery:

```python
>>> sqlglot.transpile("DATE_ADD('2020-01-01 01:01:01', 1)", "hive", "bigquery")
["DATE_ADD('2020-01-01 01:01:01', INTERVAL 1 DAY)"]
```

Hive:

<img width="432" alt="Screenshot 2023-11-22 at 4 33 40 PM" src="https://github.com/tobymao/sqlglot/assets/46752250/05d1f040-b8b1-4a2d-8ac2-8b471324b0b2">

BQ:

<img width="491" alt="Screenshot 2023-11-22 at 4 33 57 PM" src="https://github.com/tobymao/sqlglot/assets/46752250/59c72b2d-ee6b-46cc-a5b1-93d51a41f498">

In this PR we get:

```python
>>> import sqlglot
>>> sqlglot.transpile("DATE_ADD('2020-01-01 01:01:01', 1)", "hive", "bigquery")
["DATE_ADD(CAST('2020-01-01 01:01:01' AS DATETIME), INTERVAL 1 DAY)"]
```

So this now runs in BQ, though it produces a `TIMESTAMP`:

<img width="626" alt="Screenshot 2023-11-22 at 4 34 20 PM" src="https://github.com/tobymao/sqlglot/assets/46752250/6716e744-004e-4fac-857b-06302b46d4c4">